plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/actions/CodeWhispererActionPromoter.kt

```diff
@@ -7,23 +7,30 @@ import com.intellij.codeInsight.lookup.impl.actions.ChooseItemAction
 import com.intellij.openapi.actionSystem.ActionPromoter
 import com.intellij.openapi.actionSystem.AnAction
 import com.intellij.openapi.actionSystem.DataContext
-import com.intellij.openapi.editor.actionSystem.EditorAction
-import software.aws.toolkits.jetbrains.services.amazonq.CodeWhispererFeatureConfigService
-import software.aws.toolkits.jetbrains.services.codewhisperer.popup.handlers.CodeWhispererPopupLeftArrowHandler
-import software.aws.toolkits.jetbrains.services.codewhisperer.popup.handlers.CodeWhispererPopupRightArrowHandler
-import software.aws.toolkits.jetbrains.services.codewhisperer.popup.handlers.CodeWhispererPopupTabHandler
+import software.aws.toolkits.jetbrains.services.codewhisperer.service.CodeWhispererInvocationStatus
 import software.aws.toolkits.jetbrains.services.codewhisperer.service.CodeWhispererInvocationStatusNew
+import software.aws.toolkits.jetbrains.settings.CodeWhispererSettings
 
 class CodeWhispererActionPromoter : ActionPromoter {
     override fun promote(actions: MutableList<out AnAction>, context: DataContext): MutableList<AnAction> {
-        if (CodeWhispererFeatureConfigService.getInstance().getNewAutoTriggerUX()) {
-            val results = actions.toMutableList()
-            if (!CodeWhispererInvocationStatusNew.getInstance().isDisplaySessionActive()) return results
+        val results = actions.toMutableList()
+        if (!CodeWhispererInvocationStatusNew.getInstance().isDisplaySessionActive() &&
```

Comment:
qq are we using `xxxxNew` code path? 
I lose track of this code path

Reply:
This PR makes the popup UX logic to not use the xxxNew code path for many part of the logic, aka they need to be pruned, will do this in a later PR.